src/olympia/files/tasks.py

```diff
@@ -26,32 +25,30 @@ def repack_fileupload(results, upload_pk):
     # We don't trust upload.name: it's the original filename as used by the
     # developer, so it could be something else.
     if upload.path.endswith('.xpi'):
-        timer = StopWatch('files.tasks.repack_fileupload.')
-        timer.start()
-        try:
-            tempdir = tempfile.mkdtemp()  # *not* on TMP_PATH, we want local fs
-            extract_zip(upload.path, tempdir=tempdir)
-        except Exception as exc:
-            # Something bad happened, maybe we couldn't parse the zip file.
-            # @validation_task should ensure the exception is caught and
-            # transformed in a generic error message for the developer, so we
-            # just log it and re-raise.
-            log.exception(
-                'Could not extract upload %s for repack.', upload_pk, exc_info=exc
-            )
-            raise
-        timer.log_interval('1.extracted')
-        log.info('Zip from upload %s extracted, repackaging', upload_pk)
-        file_ = tempfile.NamedTemporaryFile(suffix='.zip', delete=False)
-        shutil.make_archive(os.path.splitext(file_.name)[0], 'zip', tempdir)
+        # tempdir must *not* be on TMP_PATH, we want local fs instead. It will be
+        # deleted automatically once we exit the context manager.
+        with tempfile.TemporaryDirectory(prefix='repack_fileupload_extract') as tempdir:
+            try:
+                extract_zip(upload.path, tempdir=tempdir)
+            except Exception as exc:
+                # Something bad happened, maybe we couldn't parse the zip file.
+                # @validation_task should ensure the exception is caught and
+                # transformed in a generic error message for the developer, so we
+                # just log it and re-raise.
+                log.exception(
+                    'Could not extract upload %s for repack.', upload_pk, exc_info=exc
+                )
+                raise
+            log.info('Zip from upload %s extracted, repackaging', upload_pk)
+            # We'll move the file to its final location below with move_stored_file(),
+            # so don't let tempfile delete it.
+            file_ = tempfile.NamedTemporaryFile(suffix='.zip', delete=False)
+            shutil.make_archive(os.path.splitext(file_.name)[0], 'zip', tempdir)
         with open(file_.name, 'rb') as f:
             upload.hash = 'sha256:%s' % get_sha256(f)
-        timer.log_interval('2.repackaged')
         log.info('Zip from upload %s repackaged, moving file back', upload_pk)
         move_stored_file(file_.name, upload.path)
-        timer.log_interval('3.moved')
         upload.save()
-        timer.log_interval('4.end')
```

Comment:
uh, one of my dashboards rely on these metrics...

Reply:
Ah, it didn't seem very useful, but I can add it back.

Reply:
https://github.com/mozilla/addons-server/pull/16900/commits/1459bb6ff5c05082f2440577889180326c62deea